core/src/HIP/Kokkos_HIP_KernelLaunch.hpp

```diff
@@ -170,12 +170,15 @@ struct HIPParallelLaunch<
   }
 
   static hipFuncAttributes get_hip_func_attributes() {
-    hipFuncAttributes attr;
-    hipFuncGetAttributes(
-        &attr,
-        reinterpret_cast<void const *>(
-            hip_parallel_launch_local_memory<DriverType, MaxThreadsPerBlock,
-                                             MinBlocksPerSM>));
+    static hipFuncAttributes attr = []() {
+      hipFuncAttributes attr;
```

Comment:
I am not a fan of everything being called `attr`

Reply:
How comes we don't get a `-Wshadow` warning here?

Reply:
(I am not asking to change)

Reply:
As long as we don't capture anything, I wouldn't think that this is shadowing anything but it seems that `clang` would still warn with (`Wshadow-uncaptured-local`) which we simply don't use (https://godbolt.org/z/PPx1hE).
My thinking was that we really want them to be the same so a common name makes that intention clearer. I can change if requested.